### PR TITLE
Add Datamart and features

### PR DESCRIPTION
**Pull request by seanchan-lcjg**
- add datamart for club monaco transaction, demographic and product data
- add model features

### CODE REVIEW

File: datamart.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Join Product and Sales (For sales table filtering)
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC -- validate transaction filter, same for imx crm dashboard, provided by imx crm team
```
Comment (cintiaching) [inline datamart.py:247]: should delete this comment, this logic for CM is not provided by imx crm team

File: misc.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        """
+        select
+            distinct prod_brand,
+            brand_desc
+        from
+            CMSalesProduct a
+        left join imx_prd.imx_dw_train_silver.dbo_viw_lc_xxx_brand_brand b on prod_brand = b.brand_code
+        """
+    )
+    return df
```
Comment (cintiaching) [inline misc.py:38]: `CMSalesProduct` is not defined, and see if u still need this function

File: misc.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+sales = spark.read.parquet(os.path.join(base_dir, "transaction.parquet"))
+vip = spark.read.parquet(os.path.join(base_dir, "demographic.parquet"))
+first_purchase = spark.read.parquet(os.path.join(base_dir, "first_purchase.parquet"))
+
+# COMMAND ----------
+
+sales.createOrReplaceTempView("sales")
+vip.createOrReplaceTempView("vip")
+first_purchase.createOrReplaceTempView("first_purchase")
+
```
Comment (cintiaching) [inline misc.py:15]: seems that only sales is used, kindly delete vip and first purchase

**Review by cintiaching — APPROVED** [review]
lgtm

File: misc.py (hunk 2 above)
Comment (seanchan-lcjg) [reply misc.py]: deleted

File: misc.py (hunk 3 above)
Comment (seanchan-lcjg) [reply misc.py:15]: keep for clustering use

File: datamart.py (hunk 1 above)
Comment (seanchan-lcjg) [reply datamart.py]: deleted

**Pull request merged**